src/CommunityToolkit.Maui.Analyzers/UseCommunityToolkitInitializationAnalyzer.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CommunityToolkit.Maui.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class UseCommunityToolkitInitializationAnalyzer : DiagnosticAnalyzer
+{
+	public const string DiagnosticId = "MCT001";
+
+	const string category = "Initialization";
+	static readonly LocalizableString title = new LocalizableResourceString(nameof(Resources.InitializationErrorTitle), Resources.ResourceManager, typeof(Resources));
+	static readonly LocalizableString messageFormat = new LocalizableResourceString(nameof(Resources.IniitalizationMessageFormat), Resources.ResourceManager, typeof(Resources));
+	static readonly LocalizableString description = new LocalizableResourceString(nameof(Resources.InitializationErrorMessage), Resources.ResourceManager, typeof(Resources));
+
+	static readonly DiagnosticDescriptor rule = new(DiagnosticId, title, messageFormat, category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: description);
+
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+
+	public override void Initialize(AnalysisContext context)
+	{
+		context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+		context.EnableConcurrentExecution();
+		context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ExpressionStatement);
+	}
+
+	static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+	{
+		var expressionStatement = (ExpressionStatementSyntax)context.Node;
+		var root = expressionStatement.SyntaxTree.GetRoot();
+
+		if (HasUseMauiCommunityToolkit(root))
+		{
+			return;
+		}
+
+		if (expressionStatement.ToString().Contains("UseMauiApp<")
+			&& !expressionStatement.ToString().Contains(".UseMauiCommunityToolkit("))
+		{
+			var expression = GetInvocationExpressionSyntax(expressionStatement);
+			var diagnostic = Diagnostic.Create(rule, expression.GetLocation());
+			context.ReportDiagnostic(diagnostic);
+		}
+	}
+
+	static bool HasUseMauiCommunityToolkit(SyntaxNode root)
```

Comment:
This will look into the `class` and see if some method already has `UseMauiCommunityToolkit(` method if has the user will not see any error.

Reply:
Love it!!